### PR TITLE
Rearrange tcpip.sys driver

### PR DESCRIPTION
**Pull request by Extravert-ir**
## Purpose

In order to achieve a better code organization, let's move `ip` and `lwip` libraries right into `drivers/network/tcpip` folder, as it is the only user of those libraries.

Meanwhile, remove some old networking stuff as well.

### CODE REVIEW

File: drivers/network/tcpip/ip/lwip_glue/ip.c [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-
-#include <debug.h>
+#include <lwip/netif.h>
+#include <lwip/tcpip.h>
 
 typedef struct netif* PNETIF;
 
+void
+sys_shutdown(void);
+
```
Comment (GGLinnk) [inline drivers/network/tcpip/ip/lwip_glue/ip.c:8]: Should not be here.
This is part of lwip_glue/sys_arch.h.

**Review by GGLinnk — CHANGES_REQUESTED** [review]

File: drivers/network/tcpip/lwip/src/include/arch/sys_arch.h [hunk 2]
```diff
@@ -37,7 +37,3 @@ sys_arch_protect(sys_prot_t *lev);
 
 void
 sys_arch_unprotect(sys_prot_t lev);
-
-void
-sys_shutdown(void);
-
```
[the remark this answers is not in the document]
Comment (Extravert-ir) [reply drivers/network/tcpip/lwip/src/include/arch/sys_arch.h:43]: `sys_shutdown` is not really a part of sys_arch (is it not called by lwip and not described in any documentation).
I was lazy to really merge that into lwip_glue/ip.c, but made this change so sys_arch.h contains only the stuff required for lwip

File: drivers/network/tcpip/lwip/CMakeLists.txt [hunk 3]
```diff
@@ -1,19 +1,6 @@
...
-    ${REACTOS_SOURCE_DIR}/sdk/lib/drivers/lwip/src/include/ipv4)
-
-include_directories(
-    src/include
-    src/include/ipv4)
-
 list(APPEND SOURCE
-    src/rosip.c
-    src/rostcp.c
     src/rosmem.c
```
[the remark this answers is not in the document]
Comment (Extravert-ir) [reply drivers/network/tcpip/lwip/CMakeLists.txt:3]: OK, I can move that as well

File: drivers/network/tcpip/ip/precomp.h [hunk 4]
```diff
@@ -20,5 +20,6 @@
 #include <interface.h>
 #include <ports.h>
 #include <chew.h>
+#include "lwip_glue/lwip_glue.h"
```
[the remark this answers is not in the document]
Comment (Extravert-ir) [reply drivers/network/tcpip/ip/precomp.h:23]: The usual convention is like this:
`<>` notation is for files from include folders
`""` notation is for files included by a path relative to the current file

File: drivers/network/tcpip/lwip/src/include/arch/sys_arch.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Extravert-ir) [reply drivers/network/tcpip/lwip/src/include/arch/sys_arch.h:43]: As for moving `sys_arch.h` header, I don't think that's a good idea as the header is required for lwip to compile.
So better keep `include/arch` stuff and `lwipopts.h` inside lwip folder somewhere

File: drivers/network/tcpip/include/precomp.h [hunk 5]
```diff
@@ -16,7 +16,7 @@
 #include <lock.h>
 #include <interface.h>
 #include <fileobjs.h>
-#include <chew/chew.h>
+#include "chew.h"
```
Comment (learn-more) [inline drivers/network/tcpip/include/precomp.h:19]: Chewbacca?

**Review by learn-more — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]

**Review by GGLinnk — APPROVED** [review]

**Pull request merged**